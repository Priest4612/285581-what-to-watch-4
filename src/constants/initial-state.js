export default {
  error: null,
  loading: true,
  movies: [],
  movieIdActive: null,
  genres: [],
  genreIdAcrive: null,
};
