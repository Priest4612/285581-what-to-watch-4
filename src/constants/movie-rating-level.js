const MovieRatingLevel = {
  BAD: {
    NAME: `Bad`,
    RATING: {
      MIN: 0,
      MAX: 3,
    }
  },
  NORMAL: {
    NAME: `Normal`,
    RATING: {
      MIN: 3,
      MAX: 5,
    }
  },
  GOOD: {
    NAME: `Good`,
    RATING: {
      MIN: 5,
      MAX: 8,
    }
  },
  VERY_GOOD: {
    NAME: `Very good`,
    RATING: {
      MIN: 8,
      MAX: 10,
    }
  },
  AWESOME: {
    NAME: `Awesome`,
    RATING: {
      MIN: 10,
      MAX: 10,
    }
  },
};

export {MovieRatingLevel};
