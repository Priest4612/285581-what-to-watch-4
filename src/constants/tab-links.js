const tabList = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export {tabList};
