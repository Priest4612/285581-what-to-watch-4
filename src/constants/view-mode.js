const ViewMode = {
  MAIN: `MAIN`,
  FULL: `FULL`,
};

export {ViewMode};
