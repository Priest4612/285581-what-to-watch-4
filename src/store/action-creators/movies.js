import ActionType from '../action-type';

const moviesLoaded = (movies) => ({
  type: ActionType.Movies.FETCH_MOVIES_SUCCESS,
  payload: movies,
});

const setActiveMovieId = (id) => ({
  type: ActionType.Movies.CHANGE_ACTIVE_MOVIE,
  payload: id,
});

export default {
  moviesLoaded,
  setActiveMovieId,
};
