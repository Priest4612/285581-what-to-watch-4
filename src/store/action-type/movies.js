export default {
  FETCH_MOVIES_FAILURE: `FETCH_MOVIES_FAILURE`,
  FETCH_MOVIES_SUCCESS: `FETCH_MOVIES_SUCCESS`,
  FETCH_MOVIES_REQUEST: `FETCH_MOVIES_REQUEST`,
  CHANGE_ACTIVE_MOVIE: `CHANGE_ACTIVE_MOVIE`,
};
