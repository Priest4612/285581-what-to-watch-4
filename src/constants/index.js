import {ViewMode} from './view-mode.js';
import {MovieRatingLevel} from './movie-rating-level.js';
import {tabList} from './tab-links.js';

export {
  ViewMode,
  MovieRatingLevel,
  tabList,
};
