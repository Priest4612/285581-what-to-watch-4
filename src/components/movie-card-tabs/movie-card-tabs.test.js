import React from 'react';
import renderer from 'react-test-renderer';

import {tabList} from '../../constants';
import movies from '../../mocks/test-films-mock.json';

import MovieCardTabs from './movie-card-tabs.jsx';


it(`Render MovieCardTabs -- DETAILS`, () => {
  const {rating, scoresCount, director, starring, description, runTime, genre, released} = movies[0];
  const tree = renderer
    .create(<MovieCardTabs
      activeTab={tabList.DETAILS}
      rating={rating}
      scoresCount={scoresCount}
      director={director}
      starring={starring}
      description={description}
      runTime={runTime}
      genre={genre}
      released={released}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});


it(`Render MovieCardTabs -- OVERVIEW`, () => {
  const {rating, scoresCount, director, starring, description, runTime, genre, released} = movies[2];
  const tree = renderer
    .create(<MovieCardTabs
      activeTab={tabList.DETAILS}
      rating={rating}
      scoresCount={scoresCount}
      director={director}
      starring={starring}
      description={description}
      runTime={runTime}
      genre={genre}
      released={released}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});


it(`Render MovieCardTabs -- REVIEWS`, () => {
  const {rating, scoresCount, director, starring, description, runTime, genre, released} = movies[2];
  const tree = renderer
    .create(<MovieCardTabs
      activeTab={tabList.DETAILS}
      rating={rating}
      scoresCount={scoresCount}
      director={director}
      starring={starring}
      description={description}
      runTime={runTime}
      genre={genre}
      released={released}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
