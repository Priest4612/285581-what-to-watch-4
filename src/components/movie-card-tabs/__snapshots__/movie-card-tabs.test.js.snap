// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MovieCardTabs -- DETAILS 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item
              movie-nav__item--active"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item
              false"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item
              false"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8.6
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        38 ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: Я видел его глаза. До сих пор, я никогда не видел столько тоски и боли в глазах мужчины.
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: Вуди Аллен, Мартин Скорсезе, Акира Куросава, Чарли Чаплин, Пол Верховен, Альфонсо Куарон, Джеймс Кэмерон
      </strong>
    </p>
  </div>
</div>
`;

exports[`Render MovieCardTabs -- OVERVIEW 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item
              movie-nav__item--active"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item
              false"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item
              false"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      0.1
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Bad
      </span>
      <span
        className="movie-rating__count"
      >
        823 ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: Видимо, в моих глазах было что-то такое, что продавщица как-то побледнела и даже уменьшилась в росте. Она просто смотрела на меня как мышь на удава и молчала.
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: Серджио Леоне, Фрэнк Дарабонт, Билли Уайлдер, Дэвид Кроненберг
      </strong>
    </p>
  </div>
</div>
`;

exports[`Render MovieCardTabs -- REVIEWS 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item
              movie-nav__item--active"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item
              false"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item
              false"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      0.1
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Bad
      </span>
      <span
        className="movie-rating__count"
      >
        823 ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: Видимо, в моих глазах было что-то такое, что продавщица как-то побледнела и даже уменьшилась в росте. Она просто смотрела на меня как мышь на удава и молчала.
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: Серджио Леоне, Фрэнк Дарабонт, Билли Уайлдер, Дэвид Кроненберг
      </strong>
    </p>
  </div>
</div>
`;
